pkg/filesystem/subvolume.go

```diff
@@ -407,6 +507,33 @@ func getOmapKey(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNam
 	return omapkey
 }
 
+// getSnapOmapKey gets the omap key and value details for a given snapshot.
+// Deletion of omap object required the snapshotName which is of format
+// csi.snap.snapid.
+// similarly to delete of omap key requires csi.snap.ompakey, where
+// omapkey is the snapshotcontent name which is extracted the omap object.
+func getSnapOmapKey(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, snap, fs string) string {
+
+	poolName, err := getMetadataPoolName(ctx, clientsets, OperatorNamespace, CephClusterNamespace, fs)
+	if err != nil || poolName == "" {
+		logging.Fatal(fmt.Errorf("pool name not found: %q", err))
+	}
+	snapomapval, _ := getSnapOmapVal(snap)
+
+	args := []string{"getomapval", snapomapval, "csi.snapname", "-p", poolName, "--namespace", "csi", "/dev/stdout"}
+	cmd := "rados"
+	snapshotcontentname, err := runCommand(ctx, clientsets, OperatorNamespace, CephClusterNamespace, cmd, args)
+	if err != nil || snapshotcontentname == "" {
+		logging.Info("No snapshot content found for snapshot %s: %s", snap, err)
+		return ""
+	}
+	// omap key is for format csi.volume.pvc-fca205e5-8788-4132-979c-e210c0133182
+	// hence, attaching pvname to required prefix.
+	snapomapkey := "csi.snap." + snapshotcontentname
```

Comment:
From the comment up two lines, should this prefix be "csi.volume"? Or else I don't understand the comment and why it's "csi.snap".

Reply:
my bad, thanks for noticing.